hydrator-common/src/main/java/io/cdap/plugin/common/db/DBUtils.java

```diff
@@ -297,9 +322,21 @@ public static Schema getSchema(String typeName, int sqlType, int precision, int
 
       case Types.NUMERIC:
       case Types.DECIMAL:
+        // decimal type with precision 0 is not supported
+        if (precision == 0) {
+          throw new SQLException(new UnsupportedTypeException(
+            String.format("Column %s has unsupported SQL Type: %s with precision 0.", columnName, typeName)));
+        }
         if (handleAsDecimal) {
           return Schema.decimalOf(precision, scale);
         } else {
+          //SQL DECIMAL can be 5 - 17 bytes, not all can be held in a long
+          if (precision >= 19) {
+            throw new SQLException(
+              new UnsupportedTypeException(
+                String.format("Column %s has unsupported SQL Type: %s with precision %d while trying to store decimal" +
```

Comment:
Unable to store column %s as decimal. Precision has to be less than 19.

Reply:
done :)